cilium-cli/connectivity/check/deployment.go

```diff
@@ -581,6 +668,51 @@ func (ct *ConnectivityTest) deploy(ctx context.Context) error {
 		} else {
 			ct.Info("Skipping conn-disrupt-test for NS traffic")
 		}
+
+		if ct.ShouldRunConnDisruptEgressGateway() {
+			gatewayNode, nonGatewayNode, err := ct.getGatewayAndNonGatewayNodes()
+			if err != nil {
+				return err
+			}
+			cegp := newConnDisruptCEGP(ct.params.TestNamespace, gatewayNode)
+			ct.Logf("✨ [%s] Deploying %s CiliumEgressGatewayPolicy...", ct.K8sClient().ClusterName(), cegp.Name)
+			_, err = ct.K8sClient().ApplyGeneric(ctx, cegp)
+			if err != nil {
+				return fmt.Errorf("unable to create CiliumEgressGatewayPolicy %s: %w", cegp.Name, err)
+			}
+
+			if err := ct.createTestConnDisruptServerDeployAndSvc(ctx, testConnDisruptServerEgressGatewayDeploymentName, KindTestConnDisruptEgressGateway, 1,
+				testConnDisruptEgressGatewayServiceName, testConnDisruptServerEgressGatewayAppLabel, true, newConnDisruptCNPForEgressGateway); err != nil {
+				return err
+			}
+
+			if err := ct.createTestConnDisruptClientDeployment(ctx, testConnDisruptClientEgressGatewayOnGatewayNodeDeploymentName, KindTestConnDisruptEgressGateway,
+				testConnDisruptClientEgressGatewayOnGatewayNodeAppLabel, fmt.Sprintf("test-conn-disrupt-egw.%s.svc.cluster.local.:8000", ct.params.TestNamespace),
+				1, false, map[string]string{"kubernetes.io/hostname": gatewayNode}); err != nil {
+				return err
+			}
+			if err := ct.createTestConnDisruptClientDeployment(ctx, testConnDisruptClientEgressGatewayOnNonGatewayNodeDeploymentName, KindTestConnDisruptEgressGateway,
+				testConnDisruptClientEgressGatewayOnNonGatewayNodeAppLabel, fmt.Sprintf("test-conn-disrupt-egw.%s.svc.cluster.local.:8000", ct.params.TestNamespace),
+				1, false, map[string]string{"kubernetes.io/hostname": nonGatewayNode}); err != nil {
+				return err
+			}
```

Comment:
This will not work in the context of clustermesh, because it would create the server in one cluster and the clients in the other, but cross-cluster EGW policies are not supported at the moment. Either the test should be disabled when testing against clustermesh, or client and server should be always forced to the same cluster.

Reply:
I will disable the test when testing against clustermash.